nexus/types/src/deployment.rs

```diff
@@ -969,7 +969,14 @@ impl From<BlueprintDatasetsConfig> for DatasetsConfig {
             datasets: config
                 .datasets
                 .into_iter()
-                .map(|(id, d)| (id, d.into()))
+                .filter_map(|(id, d)| {
```

Comment:
This is the actual fix - basically, converting from "Blueprint dataset" -> "Deployable dataset" format will filter out expunged datasets.

If this feels too magical, I can remove `impl From` here and instead create:

```rust
impl BlueprintDatasetsConfig {
  fn into_in_service_datasets(self) -> DatasetsConfig {
     ...
  }
}
```

(the implementation would be the same, but this could make it more explicit that we're filtering)

Reply:
I like your suggestion, seems like a good refactor :)

Reply:
Done in 0bca90ffe3023ec28fe8eb4b7d2a8620c2c91d66